pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -2303,7 +2303,7 @@ def _enumerate_with_start(
     def _get_start_value(self, node: nodes.NodeNG) -> tuple[int | None, Confidence]:
         confidence = HIGH
 
-        if isinstance(node, (nodes.Name, nodes.Call)):
+        if isinstance(node, (nodes.Name, nodes.Call, nodes.Attribute)):
             inferred = utils.safe_infer(node)
             start_val = inferred.value if inferred else None
             confidence = INFERENCE
```

Comment:
in this entire func I'm negotiating between knowing what type of node will be and crashing... would you prefer I make the else the "catch all" instead and ensure we never crash regardless of not knowing what the node could be?

Reply:
I don't really understand your point and I'm not sure I missed this in an earlier review but the broad else doesn't make a lot of sense.
`mypy` only allows it because `astroid` isn't typed and therefore every `astroid` object is `Any` as far as `mypy` can see. Doing `node.value` is not type safe sadly so we should use instance checks.

Reply:
Oftentimes if a node is not something we expect we return early. I.e. we use the pattern we talked about in https://github.com/PyCQA/pylint/issues/7787#issuecomment-1319153815. It's a way to make sure we have false negatives instead of false positive or crashes.

Reply:
Alright so... I think this would meet both of your comments?

```
    def _get_start_value(self, node: nodes.NodeNG) -> tuple[int | None, Confidence]:
        if isinstance(node, (nodes.Name, nodes.Call, nodes.Attribute)):
            inferred = utils.safe_infer(node)
            start_val = inferred.value if inferred else None
            return start_val, INFERENCE
        if isinstance(node, nodes.UnaryOp):
            return node.operand.value, HIGH
        if isinstance(node, nodes.Const):
            return node.value, HIGH

        return None, HIGH
```
?